src/main/java/core/basesyntax/service/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Inventory;
+import core.basesyntax.model.FruitTransaction;
+
+public class BalanceOperation implements FruitOperationHandler {
+    @Override
+    public void getFruitOperation(FruitTransaction transaction, Inventory inventory) {
+        inventory.getInventory().put(transaction.getFruit(), transaction.getQuantity());
```

Comment:
It is important to check the result balance before saving it to the inventory to ensure it is not negative. Please add a check to ensure that the quantity is not negative before updating the inventory.